python/cudf/cudf/core/column/column.py

```diff
@@ -820,7 +818,7 @@ def take(
 
         # TODO: For performance, the check and conversion of gather map should
         # be done by the caller. This check will be removed in future release.
-        if not is_integer_dtype(indices.dtype):
+        if indices.dtype.kind not in {"u", "i"}:
```

Comment:
Wouldn't this be a problem incase of `StructColumn` and `DecimalColumn`'s? Those types will lack `.kind` attribute.

Reply:
They have a default of `"O"` (object type), so that would work here since we want to exclude those

```python
In [1]: import cudf

In [2]: cudf.Decimal128Dtype(2)
Out[2]: Decimal128Dtype(precision=2, scale=0)

In [3]: cudf.Decimal128Dtype(2).kind
Out[3]: 'O'

In [4]: cudf.StructDtype({"a": "int64"}).kind
Out[4]: 'O'
```

Reply:
Okay, as long as that's covered we are good.